### PR TITLE
Init http3 libraries

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Add http3 libraries to enable curl build with http3 protocol support
Needed to this PR - https://github.com/NixOS/nixpkgs/pull/145610

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
сс @SuperSandro2000

**Comment by SuperSandro2000** [comment]
We can target master here and it will be pushed to staging in 6 hours.

**Comment by Izorkin** [comment]
Changed to master.

**Comment by Izorkin** [comment]
Fixed PR.

File: pkgs/development/libraries/nghttp3/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  outputs = [ "out" "dev" ];
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/ngtcp2/nghttp3";
+    description = "nghttp3 is an implementation of HTTP/3 mapping over QUIC and QPACK in C.";
+    license = licenses.mit;
+    platforms = platforms.linux;
+     maintainers = with maintainers; [ izorkin ];
```
Comment (legendofmiracles) [inline pkgs/development/libraries/nghttp3/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ izorkin ];
```

Comment (Izorkin) [reply pkgs/development/libraries/nghttp3/default.nix]: Thanks, fixed :)

**Comment by Izorkin** [comment]
Resolved conflicts.

**Comment by Izorkin** [comment]
cc @SuperSandro2000

**Comment by Izorkin** [comment]
cc @Mic92 @zowoq

File: pkgs/development/libraries/quictls/default.nix [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+, withPerl ? stdenv.hostPlatform == stdenv.buildPlatform
+}:
+
+assert (
+  lib.assertMsg (!withPerl -> stdenv.hostPlatform != stdenv.buildPlatform)
+  "withPerl should not be disabled unless cross compiling"
+);
+
+stdenv.mkDerivation rec {
+  pname = "quictls";
```
Comment (Mic92) [inline pkgs/development/libraries/quictls/default.nix:18]: Do they plan to integrate these features into openssl longterm?

File: pkgs/development/libraries/quictls/default.nix [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      echo "Found an erroneous dependency on perl ^^^" >&2
+      exit 1
+    fi
+  '';
+
+  meta = with lib; {
+    homepage = "https://quictls.github.io/openssl/";
+    description = "TLS/SSL and crypto library with QUIC APIs";
+    license = licenses.openssl;
+    platforms = platforms.all;
```
Comment (Mic92) [inline pkgs/development/libraries/quictls/default.nix:167]: Maybe?

```suggestion
    platforms = platforms.all;
    maintainers = with maintainers; [ izorkin ];
```

File: pkgs/development/libraries/quictls/default.nix (hunk 2 above)
Comment (Izorkin) [reply pkgs/development/libraries/quictls/default.nix:18]: Not anytime soon - https://daniel.haxx.se/blog/2021/10/25/the-quic-api-openssl-will-not-provide/

**Review by SuperSandro2000 — APPROVED** [review]
should be fine

**Pull request merged**